src/Illuminate/Database/Eloquent/MassPrunable.php

```diff
@@ -15,16 +15,21 @@ trait MassPrunable
      */
     public function pruneAll(int $chunkSize = 1000)
     {
+        $usesSoftDeletes = in_array(SoftDeletes::class, class_uses_recursive(get_class($this)));
```

Comment:
I don't want to make this change, but the documentation and unit tests around MassPrunable are clear that they expect it to work *even if* the query from `->prunable` doesn't explicitly select soft deleted models, soft deleted models should be hard deleted.

At least this makes it explicit here that we're fucking with the user's query, instead of it being implicit in `forceDelete` -- I hope this convinces the team to introduce a breaking change in Laravel 10 to revert my line 25-28